tools/common/Application.cs

```diff
@@ -495,21 +498,46 @@ public void RunRegistrar ()
 
 			var ps = new ReaderParameters ();
 			ps.AssemblyResolver = resolver;
-			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
-
-			var rootName = Path.GetFileNameWithoutExtension (RootAssembly);
-			if (rootName != Driver.GetProductAssembly (this))
-				throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
+			resolvedAssemblies.Add ("mscorlib", ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
+
+			var productAssembly = Driver.GetProductAssembly (this);
+			bool foundProductAssembly = false;
+			foreach (var asm in RootAssemblies) {
+				var rootName = Path.GetFileNameWithoutExtension (asm);
+				if (rootName == productAssembly)
+					foundProductAssembly = true;
+				
+				try {
+					AssemblyDefinition lastAssembly = ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse (rootName), new ReaderParameters ());
+					if (lastAssembly == null)
+						ErrorHelper.CreateWarning (7, "The root assembly '{0}' does not exist", rootName);
+					
+					if (resolvedAssemblies.TryGetValue (rootName, out var previousAssembly)) {
+						if (lastAssembly.MainModule.RuntimeVersion != previousAssembly.MainModule.RuntimeVersion) {
+							Driver.Log (2, "Attemping to load an assembly another time {0} (previous {1})", lastAssembly.FullName, previousAssembly.FullName);
+						}
+						continue;
+					}
+					
+					resolvedAssemblies.Add (rootName, lastAssembly);
+					Driver.Log (3, "Loaded {0}", lastAssembly.MainModule.FileName);
+				} catch (Exception ex) {
+					ErrorHelper.Warning (9, "Error while loading assemblies: {0}{1}{2}", rootName, Environment.NewLine, ex.ToString ());
```

Comment:
You can pass the exception as the second argument to `ErrorHelper.Warning`, that way the exception's stack trace will only be shown for verbose builds (which is what we usually do: only show stack traces for verbose builds) , so it would become:

```
ErrorHelper.Warning (9, ex, "Error while loading assemblies {0}: {1}", rootName, ex.Message);
```